EIPS/eip-7928.md

```diff
@@ -0,0 +1,199 @@
+            if address not in computed_balance_diffs:
+                computed_balance_diffs[address] = []
+            computed_balance_diffs[address].append({"tx_index": idx, "delta": delta})
+
+    # Validate access list and balance diffs
+    if not validate_access_list(block.block_access_list, computed_access_list):
+        raise InvalidBlock("Mismatch in block-level access list.")
+        
+    if not validate_balance_diffs(block.balance_diffs, computed_balance_diffs):
+        raise InvalidBlock("Mismatch in balance diffs.")
+        
+    if not validate_nonce_diffs(block.nonce_diffs, computed_nonce_diffs):
+        raise InvalidBlock("Mismatch in nonce diffs.")
+```
+
+The BAL MUST be complete and accurate. It MUST NOT contain too few entries (missing accesses) or too many entries (spurious accesses). Any missing or extra entries in the access list, balance diffs, or nonce diffs SHALL result in block invalidation.
+
+Client implementations MUST compare the accessed addresses and storage keys gathered during execution (as defined in [EIP-2929](./eip-2929.md)) with those included in the BAL to determine validity.
+
+Client implementations MAY invalidate the block right away in case any transaction steps outside the declared state.
+
+## Rationale
+
+### BAL Design Choice
+
+We chose the BAL variant for several key reasons:
+
+1. **Balance between size and parallelization benefits**: BALs enable both parallel disk reads and parallel EVM execution while keeping the size manageable.
+
+2. **Storage value inclusion**: Including post-execution values for write operations facilitates state reconstruction during syncing, enabling faster chain catch-up.
+
+3. **Balance diffs**: Including balance diffs ensures correct handling of execution that depends on account balances, crucial for parallel transaction execution.
+
+4. **Nonce diffs**: Providing pre-block nonce values for transaction senders enables validators to verify nonce ordering without sequential execution.
+
+5. **Efficiency**: An analysis of historical blocks (random sample of 1000 blocks between 22,195,599 and 22,236,441) shows that BALs would have had an average size of around 57 KiB with balance diffs adding only 9.6 KiB on average, which is a reasonable overhead given the performance benefits.
+
+6. **Transaction independence**: Analysis of the same block sample revealed that approximately 60-80% of transactions within a block are fully independent from one another, meaning they access disjoint sets of addresses and storage slots.
+
+### Block Size Considerations
+
+Including access lists increases block size, potentially impacting network propagation times. Based on analysis of historical blocks:
+
+- Average BAL size over 1,000 blocks was around 57 KiB (SSZ-encoded)
+- Average balance diff size was approximately 9.6 KiB
+- Worst-case BAL size for consuming the entire block gas limit with storage access operations would be approximately 0.93 MiB
```

Comment:
These statements are fine but should be motivated. This motivation itself could be an appendix (can include other files in EIPS and link to those! See for instance appendices/supplements of EIP 2537 (BLS precompiles): https://github.com/ethereum/EIPs/blob/master/assets/eip-2537/test-vectors.md